overlord/snapstate/snapstate.go

```diff
@@ -802,11 +802,11 @@ func FinishRestart(task *state.Task, snapsup *SnapSetup) (err error) {
 	return nil
 }
 
-// RestartSystem requests a system restart.
-// It considers how the Change the task belongs to is configured
-// (system-restart-immediate) to choose whether request an immediate
-// restart or not.
-func RestartSystem(task *state.Task, rebootInfo *boot.RebootInfo) {
+// FinishTaskWithRestart requests a system restart and sets its final
+// status.  It considers how the Change the task belongs to is
+// configured (system-restart-immediate) to choose whether request an
+// immediate restart or not.
+func FinishTaskWithRestart(task *state.Task, status state.Status, rebootInfo *boot.RebootInfo) error {
```

Comment:
this could do the setting of status on the task, no?

Reply:
Also as this doesn't cover any other cases, it should be called WithSystemRestart ?

Reply:
We could set the status of the task, but I'm not sure if I like the idea as in some tasks (say, `undoLinkSnap`) we do not always call `FinishTaskWithRestart`, so anyway we need to call `task.SetStatus` in the task - it would be maybe a bit confusing to set also in this function.

Re: WithSystemRestart I've changed things now so actually `FinishTaskWithRestart` might be called also when you just want to restart the daemon.